---
title: "N-gram analysis of biological sequences in R"
author:
 - name: "Michał Burdukiewicz"
   affiliation: [1,2]
   email: "michalburdukiewicz@gmail.com"
   contact: yes
 - name: "1. Michał Burdukiewicz; 2. Piotr Sobczyk; 3. Małgorzata Kotulska; 4. Paweł Mackiewicz"
   affiliation: [1,3]
affiliation:
 - "University of Wrocław, Department of Genomics; Wrocław University of Technology, Institute of Mathematics; Wrocław University of Technology, Department of Biomedical Engineering; University of Wrocław, Department of Genomics"
keywords: "n-gram; k-mer; feature selection; proteomics"
packages: "biogram; ranger; slam"
# bibliography: bibliography.bib
output:  
        erum::erum_abstract:
                  keep_tex: TRUE
---

N-grams (k-mers) are vectors of *n* characters derived from input
sequences, widely used in genomics, transcriptomics and proteomics.
Despite the continuous interest in the sequence analysis, there are only
a few tools tailored for comparative n-gram studies. Furthermore, the
volume of n-gram data is usually very large, making its analysis in
especially challenging.

The CRAN package ~ facilitates incorporating n-gram data in the
workflows. Aside from the efficient extraction and storage of n-grams,
the package offers also a feature selection method designed specifically
for this type of data. QuiPT (Quick Permutation Test) uses several
filtering criteria such as information gain (mutual information) to
choose significant n-grams. To speed up the computation and allow
precise estimation of small p-values, QuiPT uses analytically derived
distributions instead of a large number of permutations. In addition to
this, contains tools designed for reducing the dimensionality of the
amino acid alphabet~, further scaling down the feature space.

To illustrate the usage of n-gram data in the analysis of biological
sequences, we present two case studies performed solely in . The first,
prediction of amyloids, short proteins associated with the number of
clinical disorders as Alzheimer's or Creutzfeldt-Jakob’s diseases~,
employs random forests~ trained on n-grams. The second, detection of
signal peptides orchestrating an extracellular transport of proteins,
utilizes more complicated probabilistic framework (Hidden semi-Markov
model,) but still uses n-gram data for training.

**References**

@article{wright\_ranger:\_2015, title = {ranger: {A} {Fast}
{Implementation} of {Random} {Forests} for {High} {Dimensional} {Data}
in {C}++ and {R}}, shorttitle = {ranger}, url =
{<http://arxiv.org/abs/1508.04409>}, abstract = {We introduce the C++
application and R package ranger. The software is a fast implementation
of random forests for high dimensional data. Ensembles of
classification, regression and survival trees are supported. We describe
the implementation, provide examples, validate the package with a
reference implementation, and compare runtime and memory usage with
other implementations. The new software proves to scale best with the
number of features, samples, trees, and features tried for splitting.
Finally, we show that ranger is the fastest and most memory efficient
implementation of random forests to analyze data on the scale of a
genome-wide association study.}, urldate = {2016-03-06}, journal =
{arXiv:1508.04409 \[stat\]}, author = {Wright, Marvin N. and Ziegler,
Andreas}, month = aug, year = {2015}, note = {arXiv: 1508.04409},
keywords = {Statistics - Computation, Statistics - Machine Learning},
file = {arXiv:1508.04409
PDF:/home/michal/.zotero/zotero/hf9fdd3i.default/zotero/storage/RVAIJNCD/Wright
i Ziegler - 2015 - ranger A Fast Implementation of Random Forests
fo.pdf:application/pdf;arXiv.org
Snapshot:/home/michal/.zotero/zotero/hf9fdd3i.default/zotero/storage/6PQMGRBW/1508.html:text/html}
}

@article{murphy\_simplified\_2000, title = {Simplified amino acid
alphabets for protein fold recognition and implications for folding},
volume = {13}, issn = {1741-0126, 1741-0134}, url =
{<http://peds.oxfordjournals.org/content/13/3/149>}, doi =
{10.1093/protein/13.3.149}, abstract = {Protein design experiments have
shown that the use of specific subsets of amino acids can produce
foldable proteins. This prompts the question of whether there is a
minimal amino acid alphabet which could be used to fold all proteins. In
this work we make an analogy between sequence patterns which produce
foldable sequences and those which make it possible to detect structural
homologs by aligning sequences, and use it to suggest the possible size
of such a reduced alphabet. We estimate that reduced alphabets
containing 10–12 letters can be used to design foldable sequences for a
large number of protein families. This estimate is based on the
observation that there is little loss of the information necessary to
pick out structural homologs in a clustered protein sequence database
when a suitable reduction of the amino acid alphabet from 20 to 10
letters is made, but that this information is rapidly degraded when
further reductions in the alphabet are made.}, language = {en}, number =
{3}, urldate = {2015-10-12}, journal = {Protein Engineering}, author =
{Murphy, Lynne Reed and Wallqvist, Anders and Levy, Ronald M.}, month =
mar, year = {2000}, pmid = {10775656}, keywords = {minimal alphabet,
protein fold recognition, Sequence Alignment}, pages = {149--152}, file
= {Full Text
PDF:/home/michal/.zotero/zotero/hf9fdd3i.default/zotero/storage/AA8Q22XA/Murphy
i in. - 2000 - Simplified amino acid alphabets for protein fold
r.pdf:application/pdf;Full Text
PDF:/home/michal/.zotero/zotero/hf9fdd3i.default/zotero/storage/G6GIKP2W/Murphy
i in. - 2000 - Simplified amino acid alphabets for protein fold
r.pdf:application/pdf;Snapshot:/home/michal/.zotero/zotero/hf9fdd3i.default/zotero/storage/KVQSARMR/149.html:text/html;Snapshot:/home/michal/.zotero/zotero/hf9fdd3i.default/zotero/storage/SETRR58Q/149.html:text/html}
}

@article{fandrich\_oligomeric\_2012, series = {Amyloid {Structure},
{Function}, and {Molecular} {Mechanisms} ({Part} {II})}, title =
{Oligomeric {Intermediates} in {Amyloid} {Formation}: {Structure}
{Determination} and {Mechanisms} of {Toxicity}}, volume = {421}, issn =
{0022-2836}, shorttitle = {Oligomeric {Intermediates} in {Amyloid}
{Formation}}, url =
{<http://www.sciencedirect.com/science/article/pii/S0022283612000277>},
doi = {10.1016/j.jmb.2012.01.006}, abstract = {Oligomeric intermediates
are non-fibrillar polypeptide assemblies that occur during amyloid
fibril formation and that are thought to underlie the aetiology of
amyloid diseases, such as Alzheimer's disease, Parkinson's disease and
Huntington's disease. Focusing primarily on the oligomeric states formed
from Alzheimer's disease β-amyloid (Aβ) peptide, this review will make
references to other polypeptide systems, highlighting common principles
or sequence-specific differences. The covered topics include the
structural properties and polymorphism of oligomers, the biophysical
mechanism of peptide self-assembly and its role for pathogenicity in
amyloid disease. Oligomer-dependent toxicity mechanisms will be
explained along with recently emerging possibilities of interference.},
number = {4–5}, urldate = {2015-07-24}, journal = {Journal of Molecular
Biology}, author = {Fändrich, Marcus}, month = aug, year = {2012},
keywords = {Amyloid, intermediate, oligomer, protein folding disease,
structure}, pages = {427--440}, file = {ScienceDirect Full Text
PDF:/home/michal/.zotero/zotero/hf9fdd3i.default/zotero/storage/TDSVH6TX/Fändrich
- 2012 - Oligomeric Intermediates in Amyloid Formation
Str.pdf:application/pdf;ScienceDirect
Snapshot:/home/michal/.zotero/zotero/hf9fdd3i.default/zotero/storage/Q2NJX65V/S0022283612000277.html:text/html}
}

@book{burdukiewicz\_biogram:\_2015, title = {biogram: analysis of
biological sequences using n-grams}, url =
{<http://CRAN.R-project.org/package=biogram>}, author = {Burdukiewicz,
Michal and Sobczyk, Piotr and Lauber, Chris}, year = {2015}, note = {R
package version 1.2} }
